packages/firestore/src/core/sync_engine.ts

```diff
@@ -973,31 +940,28 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       switch (state) {
         case 'current':
         case 'not-current': {
-          return this.localStore.getNewDocumentChanges().then(
-            async changes => {
-              const synthesizedRemoteEvent = RemoteEvent.createSynthesizedRemoteEventForCurrentChange(
-                targetId,
-                state === 'current'
-              );
-              await this.emitNewSnapsAndNotifyLocalStore(
-                changes,
-                synthesizedRemoteEvent
+          try {
+            const changes = await this.localStore.getNewDocumentChanges();
+            const synthesizedRemoteEvent = RemoteEvent.createSynthesizedRemoteEventForCurrentChange(
+              targetId,
+              state === 'current'
+            );
+            await this.emitNewSnapsAndNotifyLocalStore(
+              changes,
+              synthesizedRemoteEvent
+            );
+            break;
+          } catch (error) {
+            if (isDocumentChangeMissingError(error)) {
```

Comment:
Hrm. I think the prior code was a bit more precisely correct in that it was clear that the DocumentChangeMissingError would come from the getNewDocumentChanges() call, whereas now it could (theoretically) come from createSynthesizedRemoteEventForCurrentChange() or emitNewSnapsAndNotifyLocalStore().

It's probably fine, but I'm noticing that the promise chaining code is in some ways better...  though the new code is perhaps more readable since the many line `.then(..., ...)` construct is probably easy to misread.

Anyway, this is probably good. I'm just thinking out loud.

Reply:
Yeah, I see what you mean. Would commenting the source of errors in the `catch()` block be helpful in this case?

Reply:
Probably not necessary, but wouldn't hurt (I leave it up to you. 😄 )